pkg/operator/vspherecontroller/vsphere_environment_checker.go

```diff
@@ -0,0 +1,98 @@
+package vspherecontroller
+
+import (
+	"context"
+	"math"
+	"time"
+
+	"github.com/openshift/vmware-vsphere-csi-driver-operator/pkg/operator/vspherecontroller/checks"
+	"k8s.io/apimachinery/pkg/util/wait"
+	"k8s.io/klog/v2"
+)
+
+var (
+	defaultBackoff = wait.Backoff{
+		Duration: time.Minute,
+		Factor:   2,
+		Jitter:   0.01,
+		// Don't limit nr. of steps
+		Steps: math.MaxInt32,
+		// Maximum interval between checks.
+		Cap: time.Hour * 1,
+	}
+)
+
+type vSphereEnvironmentCheckInterface interface {
+	Check(ctx context.Context, connection checks.CheckArgs) (time.Duration, checks.ClusterCheckResult, bool)
+}
+
+type VsphereEnvironmentChecker struct {
```

Comment:
nit: `VSphereEnvironmentChecker`

Reply:
Maybe I am missing something, but can it be private?

Reply:
made private.